src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -755,6 +755,13 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                             string finalRefPeFilePath = Arguments.OutputRefFilePath;
                             var refPeStreamProviderOpt = finalRefPeFilePath != null ? new CompilerEmitStreamProvider(this, finalRefPeFilePath) : null;
 
+                            RSAParameters? privateKeyOpt = null;
+                            if (compilation.Options.StrongNameProvider?.Capability == SigningCapability.SignsPeBuilder)
+                            {
+                                privateKeyOpt = compilation.StrongNameKeys.PrivateKey;
+                                // PROTOTYPE(strongname): Report an error if PrivateKey is null, meaning that they passed in a public key file.
```

Comment:
Is a warning issued in the equivalent case for legacy provider?

Reply:
in order to answer this question, I'm writing a few tests.

Reply:
Apparently we just silently fail to sign.

My implementation already does this 👍 

Removed prototype comment.